### PR TITLE
Create quality check spec for leftover debugging/development mechanisms

### PR DESCRIPTION
**Pull request by RochesterinNYC**
Example mechanisms that are screened for:
- `sleep`
- `binding.pry`
- `debugger`
- focused `rspec` tests via `fit`

- Related to discussion at https://github.com/bundler/bundler/pull/4439#issuecomment-210091297

### CODE REVIEW

File: spec/quality_spec.rb [hunk 1]
```diff
@@ -18,6 +18,18 @@ def check_for_spec_defs_with_single_quotes(filename)
     "#{filename} uses inconsistent single quotes on lines #{failing_lines.join(", ")}"
   end
 
+  def check_for_debugging_mechanisms(filename)
+    failing_lines = []
+
+    File.readlines(filename).each_with_index do |line, number|
+      failing_lines << number + 1 if line =~ /binding\.pry/ || line =~ /sleep\s*\(?\d+/ ||
```
Comment (segiddins) [inline spec/quality_spec.rb]: can we collapse this into a single regexp?

**Comment by RochesterinNYC** [comment]
Good call. Broke it into a more readable multi-line regex.

**Comment by segiddins** [comment]
@homu r+

**Comment by RochesterinNYC** [comment]
Fixed the formatting rubocop error and re-ran the flaky Travis build job.

**Comment by segiddins** [comment]
@homu r+

**Pull request merged**